pennylane/hf/tapering.py

```diff
@@ -451,16 +461,155 @@ def transform_hamiltonian(h, generators, paulix_ops, paulix_sector):
+    the Pauli-Z operator, respectively. Therefore, to get the optimal sector, for each symmetery :math:`\tau`, we first
+    define a symmetry sector by the wires it acts upon and then calculate the number of occupied orbitals (:math:`N_{OC}`)
+    present in it. If :math:`N_{OC}` is even (odd), then the correct sector for the given symmetry is
+    the one with a +1 (-1) eigenvalue.
+
+    Args:
+        qubit_op (pennylane.Hamiltonian): Hamiltonian for which symmetries are being generated to perform tapering
+        generators (list[pennylane.Hamiltonian]): list of symmetry generators for the Hamiltonian
+        active_electrons (int): The number of active electrons in the system for generating the Hartree-Fock bitstring
+
+    Returns:
+        list[int]: eigenvalues corresponding to the optimal sector which contains the ground state
+
+    .. code-block:: python
+
+        >>> symbols = ['H', 'H']
+        >>> coordinates = np.array([0., 0., -0.66140414, 0., 0., 0.66140414]))
+        >>> mol = qml.hf.Molecule(symbols, coordinates)
+        >>> H, qubits = qml.hf.generate_hamiltonian(mol)(), 4
+        >>> generators, paulix_ops = generate_symmetries(H, qubits)
+        >>> optimal_sector(H, generators, 2)
+          [1, -1, -1]
+
+    """
+
+    if active_electrons < 1:
+        raise ValueError(
+            f"The number of active electrons must be greater than zero;"
+            f"got 'electrons'={active_electrons}"
+        )
+
+    num_orbitals = len(qubit_op.wires)
+
+    if active_electrons > num_orbitals:
+        raise ValueError(
+            f"Number of active orbitals cannot be smaller than number of active electrons;"
+            f" got 'orbitals'={num_orbitals} < 'electrons'={active_electrons}."
+        )
+
+    hf_str = np.where(np.arange(num_orbitals) < active_electrons, 1, 0)
+
+    perm = []
+    for tau in generators:
+        symmstr = np.array([1 if wire in tau.wires else 0 for wire in qubit_op.wires])
+        coeff = -1 if numpy.logical_xor.reduce(numpy.logical_and(symmstr, hf_str)) else 1
+        perm.append(coeff)
+
+    return perm
+
+
+def taper_hartree_fock(num_electrons, num_wires, generators, paulix_ops, paulix_sector):
+    r"""Taper Hartree Fock according to the generated set of symmetries.
+
+    The Hartree-Fock state for a molecule is transformed to a qubit observable under Jordan-Wigner
+    transform. This observable is tapered using the same Cliffords :math:`U` that are obtained
+    from the :math:`\mathcal{Z}_2` symmetries observed in the molecular Hamiltonian. A new, tapered
+    Hartree-Fock state is built from the tapered observable by putting all the qubits which are acted
+    on by a Pauli-X or Pauli-Y operator in :math:`|1\rangle` state.
+
+    Args:
+        num_electrons (int): number of active electrons in the system for generating the Hartree-Fock bitstring
+        num_wires (int): number of wires in the system for generating the Hartree-Fock bitstring
```

Comment:
Might bring these to the end, for the sake of consistency with the other functions.